packages/components/src/Dialog/DialogContext.ts

```diff
@@ -29,10 +29,12 @@ import noop from 'lodash/noop'
 
 export interface DialogContextProps {
   closeModal: () => void
+  dialogId: string
 }
 
 const dialogContext: DialogContextProps = {
   closeModal: () => noop,
+  dialogId: '',
```

Comment:
We haven't been super consistent about prop-docs but we're _trying_ to get better. I'd also like to mark this as `@private` since I don't (at this point at least) want to encourage more use of this outside this since use-case.

```suggestion
  /**
   * Used to connect the DialogHeader's title to the Dialog's `aria-labelledby`
   * @private
   */
  id: string
```